[openimageio] /utf-8 is required on Windows, but not supplied by the cmake target

### Is your feature request related to a problem? Please describe.

If you include `<openimageio.h>` on Windows you get the following compilation error:

```
1>D:\legalize\utahcpp\openimageio\build-oiio-example-default\vcpkg_installed\x64-windows\include\OpenImageIO\detail\fmt\base.h(458,28): error C2338: static_assert failed: 'Unicode support requires compiling with /utf-8'
```

This is because openimageio depends on libfmt v11 which requires this.  However, the `OpenImageIO::OpenImageIO` target doesn't add this as a `PUBLIC` compile option for this target, so the required switch is not propagated to clients of the OpenImageIO target.

### Proposed solution

Add the necessary compile option on Windows

### Describe alternatives you've considered

There really isn't one as it prevents any portion of the library from being used.

### Additional context

See attached ZIP of source code that reproduces the problem.
[oiio-utf8.zip](https://github.com/user-attachments/files/17285355/oiio-utf8.zip)

--- a/ports/openimageio/portfile.cmake
+++ b/ports/openimageio/portfile.cmake
@@ -57,6 +57,7 @@ vcpkg_cmake_configure(
         -DUSE_TBB=OFF
         -DLINKSTATIC=OFF # LINKSTATIC breaks library lookup
         -DBUILD_MISSING_FMT=OFF
+        -DINTERNALIZE_FMT=OFF  # carry fmt's msvc utf8 usage requirements
         -DBUILD_MISSING_ROBINMAP=OFF
         -DBUILD_MISSING_DEPS=OFF
         -DSTOP_ON_WARNING=OFF

--- a/ports/openimageio/vcpkg.json
+++ b/ports/openimageio/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "openimageio",
   "version": "2.5.14.0",
+  "port-version": 1,
   "description": "A library for reading and writing images, and a bunch of related classes, utilities, and application.",
   "homepage": "https://github.com/OpenImageIO/oiio",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6646,7 +6646,7 @@
     },
     "openimageio": {
       "baseline": "2.5.14.0",
-      "port-version": 0
+      "port-version": 1
     },
     "openjpeg": {
       "baseline": "2.5.2",

--- a/versions/o-/openimageio.json
+++ b/versions/o-/openimageio.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "59dd4d494969cad63902ddaeca3dad341b918fd0",
+      "version": "2.5.14.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "7ffa8987ec86923ad9e385a26b22730ad020f79d",
       "version": "2.5.14.0",
       "port-version": 0

